### DISCUSSION TITLE
[Question][Hw6] Constraint Specification Question

### DISCUSSION THREAD

**Discussion by pingpingy1** [prosyslab-classroom/cs424-program-reasoning#305] [Q&A]
In the constraints given, is there any instance where any of the arguments for the function is not a variable?
(eg. `(<= 0 (f x 0))` or `(= (f (+ x 1)) (+ (f x) 1))`, etc.)
Also, can there be cases where the arguments are not in the order as specified in `spec.param`?
(eg. `(= (f x y) (f y x))` or `(=> (= x y) (= (f y x) 0))`, etc.)

Thank you.

**Comment by Hhro** [comment]
1. It's possible, but we won't give any hint about the evaluation test cases.
2. Yes, it's possible.

**Comment by pingpingy1** [comment]
Okay. Thank you for clarifying.
